ipywidgets/widgets/widget.py

```diff
@@ -185,11 +185,19 @@ def __del__(self):
     def open(self):
         """Open a comm to the frontend if one isn't already open."""
         if self.comm is None:
+            state, buffer_keys, buffers = self._split_state_buffers(self.get_state())
+            
             args = dict(target_name='ipython.widget',
-                        data=self.get_state())
+                        data=state,
+            )
             if self._model_id is not None:
                 args['comm_id'] = self._model_id
+            
             self.comm = Comm(**args)
+            if buffers:
+                # FIXME: workaround ipykernel missing binary message support in open-on-init
+                # send state with binary elements as second message
+                self.send_state()
```

Comment:
Eventually we will need to add support for binary messages in open on init, because sending state on comm open fixed a state race condition (of course, I failed to link to any issues in the original PR, *sigh* #84)